samples/cellular/modem_shell/src/sock/sock_shell.c

```diff
@@ -212,6 +231,104 @@ static void sock_print_usage(enum sock_shell_command command)
 	}
 }
 
+static int cmd_sock_getaddrinfo(const struct shell *shell, size_t argc, char **argv)
+{
+	int err = 0;
+
+	if (argc < 2) {
+		goto show_usage;
+	}
+
+	/* Variables for command line arguments */
+	int arg_family = AF_UNSPEC;
+	int arg_type = 0;
+	int arg_flags = 0;
+	char arg_hostname[SOCK_MAX_ADDR_LEN + 1];
+	int arg_pdn_cid = 0;
+
+	memset(arg_hostname, 0, SOCK_MAX_ADDR_LEN + 1);
+
+	optreset = 1;
+	optind = 1;
+	int opt;
+
+	while ((opt = getopt_long(argc, argv, short_options, long_options, NULL)) != -1) {
+		int addr_len = 0;
+
+		switch (opt) {
+		case 'H': /* Hostname */
+			addr_len = strlen(optarg);
+			if (addr_len > SOCK_MAX_ADDR_LEN) {
+				mosh_error(
+					"Hostname length %d exceeded. Maximum is %d.",
+					addr_len, SOCK_MAX_ADDR_LEN);
+				return -EINVAL;
+			}
+			memcpy(arg_hostname, optarg, addr_len);
+			break;
+		case 'f': /* Address family */
+			if (strcmp(optarg, "unspec") == 0) {
+				arg_family = AF_UNSPEC;
+			} else if (strcmp(optarg, "inet") == 0) {
+				arg_family = AF_INET;
+			} else if (!strcmp(optarg, "inet6")) {
+				arg_family = AF_INET6;
+			} else {
+				mosh_error(
+					"Unsupported address family=%s. Supported values are: "
+					"'unspec', 'inet' (ipv4) or 'inet6' (ipv6)",
+					optarg);
+				return -EINVAL;
+			}
+			break;
+		case 't': /* Socket type */
+			if (strcmp(optarg, "any") == 0) {
+				arg_type = 0;
+			} else if (strcmp(optarg, "stream") == 0) {
+				arg_type = SOCK_STREAM;
+			} else if (strcmp(optarg, "dgram") == 0) {
+				arg_type = SOCK_DGRAM;
+			} else {
+				mosh_error(
+					"Unsupported address type=%s. Supported values are: "
+					"'any', 'stream' (tcp) or 'dgram' (udp)",
+					optarg);
+				return -EINVAL;
+			}
+			break;
+		case 'I': /* PDN CID */
+			arg_pdn_cid = atoi(optarg);
+			if (arg_pdn_cid <= 0) {
```

Comment:
That needs to be addressed throughout the file, I see other occurrences for example in `cmd_sock_connect`.

Reply:
@jhirsi Can you have a look? Isn't CID 0 the default PDP context?

Reply:
CID 0 is the default PDP context, and providing `-I 0` should be the same as not provide a CID.

For testing purpose (and for completeness) it could be ok to allow this. I can change the existing check in connect and update this.

Reply:
Thanks, I agree that it would be good to allow also the default PDP context to be specified, even though the same functionality can be achieved without the parameter. `modem_shell` is basically a test application, so it should be possible to test this kind of scenarios. If the application is using more than one PDP context, I don't think it's far fetched to use CID 0 for traffic targeted to the default PDP context.

Reply:
Just note that atoi returns zero for invalid values so `-I hey` will be same as `-I 0`.

Reply:
The same issue exists for many arguments in `sock_shell.c`, including socket_id.
This should be fixed in a separate PR.